src/data/real/irrational.lean

```diff
@@ -257,4 +257,52 @@ open irrational
 @[simp] theorem irrational_inv_iff : irrational x⁻¹ ↔ irrational x :=
 ⟨of_inv, irrational.inv⟩
 
+/-!
+### Lemmas about Irrationals between Reals
+-/
+
+/-- An irrational number between any two distinct rational numbers.-/
+noncomputable def irrational_btwn_rats {x y : ℚ} (h : x < y) : { r : ℝ // irrational r } :=
```

Comment:
I think a more mathlib-style approach here would be to split this into two definitions, `irrational_btwn_rats : ℝ` and `irrational_btwn_rats_irrational : irrational (irrational_btwn_rats h)`.